components/odh-notebook-controller/controllers/notebook_runtime.go

```diff
@@ -0,0 +1,238 @@
+	configMap := &corev1.ConfigMap{}
+	err := r.Get(ctx, types.NamespacedName{Name: configMapName, Namespace: namespace}, configMap)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			return nil, false, nil
+		}
+		return nil, false, err
+	}
+	return configMap, true, nil
+}
+
+func (r *OpenshiftNotebookReconciler) syncRuntimeImagesConfigMap(ctx context.Context, notebookNamespace string, controllerNamespace string, config *rest.Config) error {
+	log := r.Log.WithValues("namespace", notebookNamespace)
+
+	// Create a dynamic client
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Error(err, "Error creating dynamic client")
+		return err
+	}
+
+	// Define GroupVersionResource for ImageStreams
+	ims := schema.GroupVersionResource{
+		Group:    "image.openshift.io",
+		Version:  "v1",
+		Resource: "imagestreams",
+	}
+
+	// Fetch ImageStreams from controllerNamespace namespace
+	imagestreams, err := dynamicClient.Resource(ims).Namespace(controllerNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Error(err, "Failed to list ImageStreams", "Namespace", controllerNamespace)
+		return err
+	}
+
+	// Prepare data for ConfigMap
+	data := make(map[string]string)
+	for _, item := range imagestreams.Items {
+		labels := item.GetLabels()
+		if labels["opendatahub.io/runtime-image"] == "true" {
+			tags, found, err := unstructured.NestedSlice(item.Object, "spec", "tags")
+			if err != nil || !found {
+				log.Error(err, "Failed to extract tags from ImageStream", "ImageStream", item.GetName())
+				continue
+			}
+
+			for _, tag := range tags {
+				tagMap, ok := tag.(map[string]interface{})
+				if !ok {
+					continue
+				}
+
+				// Extract metadata annotation
+				annotations, found, err := unstructured.NestedMap(tagMap, "annotations")
+				if err != nil || !found {
+					annotations = map[string]interface{}{}
+				}
+
+				metadataRaw, ok := annotations["opendatahub.io/runtime-image-metadata"].(string)
+				if !ok || metadataRaw == "" {
+					metadataRaw = "[]"
+				}
+
+				// Parse metadata
+				metadataParsed := parseRuntimeImageMetadata(metadataRaw)
+				displayName := extractDisplayName(metadataParsed)
+
+				// Construct the key name
+				if displayName != "" {
+					formattedName := formatKeyName(displayName)
+					data[formattedName] = metadataParsed
+				}
+			}
+		}
+	}
+
+	// Create or update the ConfigMap in the Notebook's namespace
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: notebookNamespace,
+			Labels:    map[string]string{"opendatahub.io/managed-by": "workbenches"},
+		},
+		Data: data,
+	}
+
+	// Check if the ConfigMap exists
+	existingConfigMap, configMapExists, err := r.getRuntimeConfigMap(ctx, configMapName, notebookNamespace)
+	if err != nil {
+		log.Error(err, "Error getting ConfigMap", "ConfigMap.Name", configMapName)
+		return err
+	}
+
+	if configMapExists {
```

Comment:
This part where, I would like more thoughts,
if the data is empty, should we be creating the configmap ?
as it doesn't cause any new content, it doesn't matter. 
though do you folks think of any other cases?

Reply:
Yes, that makes sense. I reviewed this scenario, and if we proceed this way, we should apply the following changes to the functionality of the MountPipelineRuntimeImages function when req.Operation == admissionv1.Update to ensure the ConfigMap is updated accordingly.

Specifically, if the system starts without a ConfigMap, it should detect when the ConfigMap is later created and update the pod specifications accordingly.

This commit applies your suggestion:  https://github.com/opendatahub-io/kubeflow/pull/513/commits/758968ceb3bd74d54537f125f3d2aa9d2b6fbd87 

Could you take a look once you are available on these new changes?

Of course, we also need to update the tests.

Reply:
> Specifically, if the system starts without a ConfigMap, it should detect when the ConfigMap is later created and update the pod specifications accordingly.

creation of the configmap should restart running workbenches?

Reply:
> > Specifically, if the system starts without a ConfigMap, it should detect when the ConfigMap is later created and update the pod specifications accordingly.
> 
> creation of the configmap should restart running workbenches?

Not Creation. Update the notebook's specs sorry

Reply:
> I was meaning the to update the mounting on the notebook's specs

that restarts the notebook pod, I fear

Reply:
ah, it would restart notebook pod, but it actually then would get blocked in

https://github.com/opendatahub-io/kubeflow/blob/d427a2d3eb0fc1b55b2038df4061f5b560eba9ea/components/odh-notebook-controller/controllers/notebook_webhook.go#L399-L400